praw/exceptions.py

```diff
@@ -1,38 +1,96 @@
 """PRAW exception classes.
 
-Includes two main exceptions: :class:`.APIException` for when something goes
-wrong on the server side, and :class:`.ClientException` when something goes
-wrong on the client side. Both of these classes extend :class:`.PRAWException`.
+Includes two main exceptions: :class:`.RedditAPIException` for when something
+goes wrong on the server side, and :class:`.ClientException` when something
+goes wrong on the client side. Both of these classes extend
+:class:`.PRAWException`.
 
 All other exceptions are subclassed from :class:`.ClientException`.
 
 """
-from typing import Optional
+from typing import List, Optional, TypeVar, Union
+
+_RedditErrorItem = TypeVar("_RedditErrorItem")
 
 
 class PRAWException(Exception):
     """The base PRAW Exception that all other exception classes extend."""
 
 
-class APIException(PRAWException):
-    """Indicate exception that involve responses from Reddit's API."""
+class RedditErrorItem:
+    """Represents a single error returned from Reddit's API."""
+
+    @property
+    def error_message(self):
+        """Get the completed error message string."""
+        error_str = "{}: '{}'".format(self.error_type, self.message)
```

Comment:
I know you copied this from how it was before. Nevertheless, I think it might be a bit more robust to use `repr(self.message)` instead of manually putting single quotes in the string.